src/App.jsx

```diff
@@ -14,45 +17,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const IS_SORTED_ALPHABET = 'A';
+const IS_SORTED_LENGTH = 'L';
+
+function getPreparedGoods(goods, { isSorted, isReversed, query }) {
+  const preparedGoods = [...goods];
+
+  switch (isSorted) {
+    case IS_SORTED_ALPHABET:
+      preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case IS_SORTED_LENGTH:
+      preparedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+
+      break;
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [isSorted, setIsSorted] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer,
+    { isSorted, isReversed });
+
+  const reset = () => {
+    setIsSorted('');
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': isSorted !== IS_SORTED_ALPHABET,
+          })}
+          onClick={() => setIsSorted(IS_SORTED_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': isSorted !== IS_SORTED_LENGTH,
+          })}
+          onClick={() => setIsSorted(IS_SORTED_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
+        >
+          Reverse
+        </button>
+
+        {(isSorted !== '' || isReversed) && (
```

Comment:
a superficial comparison is used here and if you have a non-empty string, it will be automatically converted to `true`
```suggestion
        {(isSorted || isReversed) && (
```